tests/tools/test_dist_part.py

```diff
@@ -0,0 +1,226 @@
+    for feat in ['feat', 'label', 'year']:
+        for i in range(num_chunks):
+            chunk_f_name = '{}-{}.npy'.format(feat, i)
+            chunk_f_name = os.path.join(output_node_data_dir, chunk_f_name)
+            assert os.path.isfile(chunk_f_name)
+            feat_array = np.load(chunk_f_name)
+            assert feat_array.shape[0] == num_papers // num_chunks
+
+    # check edge_data
+    num_edges = {
+        'paper:cites:paper': num_cite_edges,
+        'author:writes:paper': num_write_edges,
+        'paper:rev_writes:author': num_write_edges
+    }
+    output_edge_data_dir = os.path.join(output_dir, 'edge_data')
+    for etype, feat in [
+        ['paper:cites:paper', 'count'],
+        ['author:writes:paper', 'year'],
+        ['paper:rev_writes:author', 'year']
+    ]:
+        output_edge_sub_dir = os.path.join(output_edge_data_dir, etype)
+        for i in range(num_chunks):
+            chunk_f_name = '{}-{}.npy'.format(feat, i)
+            chunk_f_name = os.path.join(output_edge_sub_dir, chunk_f_name)
+            assert os.path.isfile(chunk_f_name)
+            feat_array = np.load(chunk_f_name)
+            assert feat_array.shape[0] == num_edges[etype] // num_chunks
+
+def test_partition():
+    # Step1: graph partition
+    output_dir = '2parts'
+    os.system('python tools/partition_algorithms/random_partition.py '\
+              '--metadata chunked-data/metadata.json --output_path {} --num_partitions 2'.format(output_dir))
+    for ntype in ['author', 'institution', 'paper']:
+        fname = os.path.join(output_dir, '{}.txt'.format(ntype))
+        with open(fname, 'r') as f:
+            header = f.readline().rstrip()
+            assert isinstance(int(header), int)
+
+def test_dispatch():
+    # Step2: data dispatch
+    out_dir = 'partitioned'
+    num_parts = 2
+    with open('ip_config.txt', 'w') as f:
+        f.write('127.0.0.1\n')
+        f.write('127.0.0.2\n')
+
+    os.system('python tools/dispatch_data.py --in-dir chunked-data '\
+              '--partitions-dir 2parts --out-dir {} --ip-config ip_config.txt'.format(out_dir))
+
+    # check metadata.json
+    meta_fname = os.path.join(out_dir, 'metadata.json')
+    with open(meta_fname, 'rb') as f:
+        meta_data = json.load(f)
+
+    all_etypes = ['affiliated_with', 'writes', 'cites', 'rev_writes']
+    for etype in all_etypes:
+        assert len(meta_data['edge_map'][etype]) == num_parts
+    assert meta_data['etypes'].keys() == set(all_etypes)
+    assert meta_data['graph_name'] == 'mag240m'
+
+    all_ntypes = ['author', 'institution', 'paper']
+    for ntype in all_ntypes:
+        assert len(meta_data['node_map'][ntype]) == num_parts
+    assert meta_data['ntypes'].keys() == set(all_ntypes)
+    assert meta_data['num_edges'] == 4200
+    assert meta_data['num_nodes'] == 720
+    assert meta_data['num_parts'] == num_parts
+    assert meta_data['part_method'] == 'metis'
```

Comment:
why it's `metis`, did we partition randomly?

Reply:
I think it's the default value used [here](https://github.com/dmlc/dgl/blob/dist_part/tools/distpartitioning/convert_partition.py#L249). It probably does not make sense, then I can remove it first.

Reply:
Removed.